lib/matplotlib/dates.py

```diff
@@ -286,14 +286,22 @@ def _from_ordinalf(x, tz=None):
 
     remainder = float(x) - ix
 
-    # Round down to the nearest microsecond.
-    dt += datetime.timedelta(microseconds=int(remainder * MUSECONDS_PER_DAY))
+    # Round to the nearest microsecond.
+    dt += datetime.timedelta(
```

Comment:
This could be an `else` of the `if` below, couldn't it?

Reply:
Then the code below would need to change. The above is needed to set the hours, minutes, and seconds; the stuff below changes only microseconds.